### ISSUE TITLE
Connectivity.isOnlineNow throws security exception

### ISSUE THREAD

**Issue by syarihu** [plaidev/karte-android-sdk#14]
## Expected Behavior

This was confirmed in the crash report of an app that uses KARTE SDK.
This appears to be a bug in the Android Platform.
refs: https://issuetracker.google.com/issues/175055271

So I would like you to use runCatching or something similar to prevent crashes from occurring.

## Current Behavior

Android 11 crashes with a security exception when Connectivity.isOnlineNow is called.

<img src="https://user-images.githubusercontent.com/5127846/129017403-92b91f47-0838-4e03-8b36-6b6a4b5da131.png" width=300 /><img src="https://user-images.githubusercontent.com/5127846/129017407-00466445-2b28-4348-bbf6-925a9db4e93c.png" width=300 />
<img src="https://user-images.githubusercontent.com/5127846/129017411-7fc7b9c3-9403-42db-83e5-5e4d09202098.png" width=300 /><img src="https://user-images.githubusercontent.com/5127846/129017412-856ffaed-36da-4d7a-88e4-c1d4981eaa78.png" width=300 /><img src="https://user-images.githubusercontent.com/5127846/129017414-9d6ae293-fdcb-4cfa-a056-111b2c531bd9.png" width=300 />

## Steps to Reproduce

I tried running the following code on my Xperia 1 II (Android 11) , but the crash was not reproduced at hand.

```kt
val manager =
    context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
manager.getNetworkCapabilities(manager.activeNetwork)
    ?.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) ?: false

```

## SDK Version

- core 2.11.0

## Android Version

- Android 11

## Build Environment

- Android Studio Version Arctic Fox | 2020.3.1
- Gradle Version 7.0.2
- Android Gradle Plugin Version 7.0.0

## Context Details

- More than a dozen crash reports come up a day but I don't known user impact

## Logs

```
Fatal Exception: java.lang.SecurityException: Package android does not belong to 10158
       at android.os.Parcel.createExceptionOrNull(Parcel.java:2373)
       at android.os.Parcel.createException(Parcel.java:2357)
       at android.os.Parcel.readException(Parcel.java:2340)
       at android.os.Parcel.readException(Parcel.java:2282)
       at android.net.IConnectivityManager$Stub$Proxy.getNetworkCapabilities(IConnectivityManager.java:2456)
       at android.net.ConnectivityManager.getNetworkCapabilities(ConnectivityManager.java:1385)
       at io.karte.android.utilities.connectivity.Connectivity.isOnlineNow(Connectivity.java:50)
       at io.karte.android.utilities.connectivity.Connectivity.isOnline(Connectivity.java:34)
       at io.karte.android.tracking.queue.Dispatcher.dequeue(Dispatcher.java:104)
       at io.karte.android.tracking.queue.Dispatcher.access$enqueue(Dispatcher.java:48)
       at io.karte.android.tracking.queue.Dispatcher$push$2.invoke(Dispatcher.java:79)
       at io.karte.android.tracking.queue.Dispatcher$push$2.invoke(Dispatcher.java:48)
       at io.karte.android.tracking.queue.DispatcherKt$sam$java_lang_Runnable$0.run(DispatcherKt.java:2)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:223)
       at android.os.HandlerThread.run(HandlerThread.java:67)
```

## Possible Implementation

https://github.com/plaidev/karte-android-sdk/blob/1d6574570e99a3778a48d2443fa7e4aaa75cce85/core/src/main/java/io/karte/android/utilities/connectivity/Connectivity.kt#L47-L53

```kt
  @RequiresApi(Build.VERSION_CODES.M)
  private fun isOnlineNow(context: Context): Boolean {
    return kotlin.runCatching {
          val manager = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
          manager.getNetworkCapabilities(manager.activeNetwork)
              ?.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
      }.onFailure { e ->
          Logger.e(LOG_TAG, "Failed to getNetworkCapabilities: $e")
      }.getOrNull() ?: false
  }
```

**Comment by wasnot** [comment]
@syarihu Thank you for reporting.

We were already recognized and fixed `SecurityException` that occurred in ConnectivityManager.
(Probably it was a report from your company)
However, as you pointed out, there seems to be an omission in this correction.
very sorry.

As mentioned in the issue you notified, similar fixes have been made in other SDKs, and this discussion points out the possibility of behavior due to the Android work profile management.
https://github.com/microsoft/cpp_client_telemetry/pull/798
(However, we have not verified it and reproduced it.)

We will adopt the proposed correction.
Please wait for a while until the release.

**Comment by syarihu** [comment]
Thank you for adopting my suggestion.
I'm looking forward to the release.

**Comment by tikidunpon** [comment]
@syarihu 
We have released core 2.12.1, which fixes this issue, please update your dependencies.
https://developers.karte.io/docs/release-notes-android-sdk-v2#releases---20210820

Thank you for reporting.

**Comment by syarihu** [comment]
Thank you for release!
